pkg_resources/__init__.py

```diff
@@ -419,7 +419,7 @@ def _macos_vers(_cache=[]):
                     if 'ProductVersion' in plist_content:
                         version = plist_content['ProductVersion']
 
-        _cache.append(version.split('.'))
+        _cache = [version.split('.')]
```

Comment:
This would be a change in the intent of the implementation wouldn't it?
Wasn't this function "over-cleverly" written so that the `_cache` argument is modified by the first call and then it actually works as a cache on subsequent calls?

The `_` at the beginning of the argument name gives me the impression that this function is never supposed to be invoked with an explicit argument and that the mutable default argument is intentional...

To be properly re-written, we would have to use something like `functools.cache` (or `lru_cache(maxsize=None)` for Python < 3.9) I guess...

Reply:
You're right of course.

Alternatively `_cache` could be moved outside the function:
```python
_cache = []

def _macos_vers():
    if not _cache:
        ...
```

Reply:
If that is the selected approach, we should use a more specific name.

Alternatively we can use something like:

```python
@functools.lru_cache(maxsize=None)
def _macos_vers():
    version = platform.mac_ver()[0]
    # fallback for MacPorts
    if version == '':
        plist = '/System/Library/CoreServices/SystemVersion.plist'
        if os.path.exists(plist):
            if hasattr(plistlib, 'readPlist'):
            plist_content = plistlib.readPlist(plist)
                if 'ProductVersion' in plist_content:
                    version = plist_content['ProductVersion']
    return version.split('.')
```

and get rid of the variable all together (similar approach for the other overly-clever function).

Reply:
By the way, function [`plistlib.readPlist`](https://docs.python.org/3.8/library/plistlib.html#plistlib.readPlist) has been [removed from Python  ≥ 3.9:](https://github.com/python/cpython/pull/15615)
> Deprecated since version 3.4: Use [load()](https://docs.python.org/3.8/library/plistlib.html#plistlib.load) instead.

Reply:
The cache mechanism is now based on `@functools.lru_cache`.